pkg/config/app/kuma-cp/kuma-cp.defaults.yaml

```diff
@@ -43,6 +43,8 @@ store:
 bootstrapServer:
   # Port of Server that provides bootstrap configuration for dataplanes
   port: 5682 # ENV: KUMA_BOOTSTRAP_SERVER_PORT
+  # Public URL to reach Bootstrap server. ex: https://bootstrap.kuma.com:1234, its autoconfigured if blank
+  publicUrl: # ENV: KUMA_BOOTSTRAP_SERVER_PUBLIC_URL
```

Comment:
See `CatalogConfig`:
```
type BootstrapApiConfig struct {
	Url string
}

type DataplaneTokenApiConfig struct {
	LocalUrl  string
	PublicUrl string
}

type AdminApiConfig struct {
	LocalUrl  string
	PublicUrl string
}

type MonitoringAssignmentApiConfig struct {
	Url string
}
```

I think instead of introducing publicUrl in every server it's better to include CatalogConfig to YAML and ENV config. this way you can override the catalog fields with your public URLs.

Reply:
yeah, i can do that..

Reply:
i can make changes for bootstrap and admin url.. how about sds.. i am guessing its going to sit with sdsServer config..

Also if i move the config to catalogConfig it may not very intuitive to the end user as catalog is an internal concept and not exposed to them yet..

Reply:
If you expose it, the catalog will be just a part of Kuma Config.

About XDS/SDS. Why not just set the `KUMA_GENERAL_ADVERTISED_HOSTNAME` to the address of your reverse proxy for Kuma CP? For API Server/Bootstrap/Admin it didn't work because the change of `http://` -> `https://`, but I think here it should be ok.